flake8_trio.py

```diff
@@ -592,12 +624,10 @@ def visit_AsyncFunctionDef(self, node: ast.AsyncFunctionDef):
     def check_function_exit(self, node: Union[ast.Return, ast.AsyncFunctionDef]):
         # error if function exits w/o guaranteed checkpoint since function entry
         method = "return" if isinstance(node, ast.Return) else "exit"
+        error_code = "TRIO108" if self.yield_count else "TRIO107"
```

Comment:
Small perf optimization, but it's equally clear code: move this inside the if-statement, so we only compute it in the (relatively rare) cases where we actually use the results.